<h1>Activity 1 - PALMER PENGUINS DATA ANALYTICS PROJECT </h1>
<hr>
<h3>Analyst : Willmyr Kurt M. Verastigue<h3>

In [270]:
#import libraries
import numpy as np
import pandas as pd 
import warnings
warnings.filterwarnings('ignore')

<h2>Converting Dataset(s) into DataFrame</h2>

<h2>REQUIREMENTS</h2>
<ul>
    <li>
        Name of the DataFrame
    </li>
    <li>
        File Path
    </li>
</ul>

In [271]:
penguins = pd.read_csv('Datasets\\penguins.csv')

In [272]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

<h2>DESCRIPTIVE ANALYTICS</h2>

In [273]:
penguins.describe()

rowid  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count  344.000000      342.000000     342.000000         342.000000   
mean   172.500000       43.921930      17.151170         200.915205   
std     99.448479        5.459584       1.974793          14.061714   
min      1.000000       32.100000      13.100000         172.000000   
25%     86.750000       39.225000      15.600000         190.000000   
50%    172.500000       44.450000      17.300000         197.000000   
75%    258.250000       48.500000      18.700000         213.000000   
max    344.000000       59.600000      21.500000         231.000000   

       body_mass_g         year  
count   342.000000   344.000000  
mean   4201.754386  2008.029070  
std     801.954536     0.818356  
min    2700.000000  2007.000000  
25%    3550.000000  2007.000000  
50%    4050.000000  2008.000000  
75%    4750.000000  2009.000000  
max    6300.000000  2009.000000

In [274]:
penguins.describe().T

count         mean         std     min       25%      50%  \
rowid              344.0   172.500000   99.448479     1.0    86.750   172.50   
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                       75%     max  
rowid               258.25   344.0  
bill_length_mm       48.50    59.6  
bill_depth_mm        18.70    21.5  
flipper_length_mm   213.00   231.0  
body_mass_g        4750.00  6300.0  
year               2009.00  2009.0

<h2>CHECKING FOR NULL VALUES</h2>

In [275]:
penguins.isnull()

rowid  species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    False    False   False           False          False              False   
1    False    False   False           False          False              False   
2    False    False   False           False          False              False   
3    False    False   False            True           True               True   
4    False    False   False           False          False              False   
..     ...      ...     ...             ...            ...                ...   
339  False    False   False           False          False              False   
340  False    False   False           False          False              False   
341  False    False   False           False          False              False   
342  False    False   False           False          False              False   
343  False    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 9 columns]

In [276]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2>FILTERING A COLUMN</h2>

In [277]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [278]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [279]:
penguins['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

<h2>OPERATORS</h2>

In [280]:
penguins[penguins['species'] == 'Chinstrap']

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

In [281]:
penguins[penguins['species'] != 'Chinstrap']

rowid species     island  bill_length_mm  bill_depth_mm  \
0        1  Adelie  Torgersen            39.1           18.7   
1        2  Adelie  Torgersen            39.5           17.4   
2        3  Adelie  Torgersen            40.3           18.0   
3        4  Adelie  Torgersen             NaN            NaN   
4        5  Adelie  Torgersen            36.7           19.3   
..     ...     ...        ...             ...            ...   
271    272  Gentoo     Biscoe             NaN            NaN   
272    273  Gentoo     Biscoe            46.8           14.3   
273    274  Gentoo     Biscoe            50.4           15.7   
274    275  Gentoo     Biscoe            45.2           14.8   
275    276  Gentoo     Biscoe            49.9           16.1   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
271                NaN          NaN     NaN  2009  
272              215.0       4850.0  female  2009  
273              222.0       5750.0    male  2009  
274              212.0       5200.0  female  2009  
275              213.0       5400.0    male  2009  

[276 rows x 9 columns]

<h2>SEPARATING DATA FRAMES</h2>

<h3> CHINSTRAP DATA FRAMES </h3>


In [282]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']
chinstrap_df

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

<h3> ADELIE DATA FRAMES </h3>


In [283]:
adelie_df = penguins[penguins['species'] == 'Adelie']
adelie_df

rowid species     island  bill_length_mm  bill_depth_mm  \
0        1  Adelie  Torgersen            39.1           18.7   
1        2  Adelie  Torgersen            39.5           17.4   
2        3  Adelie  Torgersen            40.3           18.0   
3        4  Adelie  Torgersen             NaN            NaN   
4        5  Adelie  Torgersen            36.7           19.3   
..     ...     ...        ...             ...            ...   
147    148  Adelie      Dream            36.6           18.4   
148    149  Adelie      Dream            36.0           17.8   
149    150  Adelie      Dream            37.8           18.1   
150    151  Adelie      Dream            36.0           17.1   
151    152  Adelie      Dream            41.5           18.5   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
147              184.0       3475.0  female  2009  
148              195.0       3450.0  female  2009  
149              193.0       3750.0    male  2009  
150              187.0       3700.0  female  2009  
151              201.0       4000.0    male  2009  

[152 rows x 9 columns]

<h3> GENTOO DATA FRAMES </h3>


In [284]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_df

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152    153  Gentoo  Biscoe            46.1           13.2              211.0   
153    154  Gentoo  Biscoe            50.0           16.3              230.0   
154    155  Gentoo  Biscoe            48.7           14.1              210.0   
155    156  Gentoo  Biscoe            50.0           15.2              218.0   
156    157  Gentoo  Biscoe            47.6           14.5              215.0   
..     ...     ...     ...             ...            ...                ...   
271    272  Gentoo  Biscoe             NaN            NaN                NaN   
272    273  Gentoo  Biscoe            46.8           14.3              215.0   
273    274  Gentoo  Biscoe            50.4           15.7              222.0   
274    275  Gentoo  Biscoe            45.2           14.8              212.0   
275    276  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 9 columns]

<h3> HOW MANY CHINSTRAP IS WEIGHTING OVER 4000G </h3>

In [285]:
penguins['species'] == 'Chinstrap'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [286]:
penguins['body_mass_g'] >= 4000

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [287]:
chinstrap_heavy = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] <= 4000)]
chinstrap_heavy

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
281    282  Chinstrap  Dream            45.2           17.8   
282    283  Chinstrap  Dream            46.1           18.2   
283    284  Chinstrap  Dream            51.3           18.2   
285    286  Chinstrap  Dream            51.3           19.9   
286    287  Chinstrap  Dream            46.6           17.8   
287    288  Chinstrap  Dream            51.7           20.3   
288    289  Chinstrap  Dream            47.0           17.3   
290    291  Chinstrap  Dream            45.9           17.1   
292    293  Chinstrap  Dream            50.3           20.0   
293    294  Chinstrap  Dream            58.0           17.8   
294    295  Chinstrap  Dream            46.4           18.6   
296    297  Chinstrap  Dream            42.4           17.3   
297    298  Chinstrap  Dream            48.5           17.5   
298    299  Chinstrap  Dream            43.2           16.6   
299    300  Chinstrap  Dream            50.6           19.4   
300    301  Chinstrap  Dream            46.7           17.9   
302    303  Chinstrap  Dream            50.5           18.4   
303    304  Chinstrap  Dream            49.5           19.0   
304    305  Chinstrap  Dream            46.4           17.8   
306    307  Chinstrap  Dream            40.9           16.6   
308    309  Chinstrap  Dream            42.5           16.7   
310    311  Chinstrap  Dream            49.7           18.6   
311    312  Chinstrap  Dream            47.5           16.8   
312    313  Chinstrap  Dream            47.6           18.3   
314    315  Chinstrap  Dream            46.9           16.6   
316    317  Chinstrap  Dream            49.0           19.5   
317    318  Chinstrap  Dream            46.2           17.5   
318    319  Chinstrap  Dream            50.9           19.1   
319    320  Chinstrap  Dream            45.5           17.0   
320    321  Chinstrap  Dream            50.9           17.9   
322    323  Chinstrap  Dream            50.1           17.9   
324    325  Chinstrap  Dream            51.5           18.7   
325    326  Chinstrap  Dream            49.8           17.3   
326    327  Chinstrap  Dream            48.1           16.4   
327    328  Chinstrap  Dream            51.4           19.0   
328    329  Chinstrap  Dream            45.7           17.3   
330    331  Chinstrap  Dream            42.5           17.3   
331    332  Chinstrap  Dream            52.2           18.8   
332    333  Chinstrap  Dream            45.2           16.6   
334    335  Chinstrap  Dream            50.2           18.8   
335    336  Chinstrap  Dream            45.6           19.4   
336    337  Chinstrap  Dream            51.9           19.5   
337    338  Chinstrap  Dream            46.8           16.5   
338    339  Chinstrap  Dream            45.7           17.0   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
281              198.0       3950.0  female  2007  
282              178.0       3250.0  female  2007  
283              197.0       3750.0    male  2007  
285              198.0       3700.0    male  2007  
286              193.0       3800.0  female  2007  
287              194.0       3

In [288]:
chinstrap_heavy.info()

<class 'pandas.core.frame.DataFrame'>
Index: 53 entries, 276 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   rowid              53 non-null     int64  
 1   species            53 non-null     object 
 2   island             53 non-null     object 
 3   bill_length_mm     53 non-null     float64
 4   bill_depth_mm      53 non-null     float64
 5   flipper_length_mm  53 non-null     float64
 6   body_mass_g        53 non-null     float64
 7   sex                53 non-null     object 
 8   year               53 non-null     int64  
dtypes: float64(4), int64(2), object(3)
memory usage: 4.1+ KB


<h2> FILLING UP MISSING VALUES </h2>

In [289]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [290]:
penguins['bill_length_mm'].isnull().sum()

np.int64(2)

In [291]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [292]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [293]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [294]:
penguins['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [295]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [296]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [297]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [298]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [299]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [300]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [301]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [302]:
penguins.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [303]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [304]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [305]:
penguins.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [306]:
#penguins_sex = penguins[penguins['sex'] == 'male']
#penguins_sex

<h2>CHANGING DATA TYPES</h2>

In [307]:
penguins

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year  
0           181.000000  3750.000000    male  2007  
1           186.000000  3800.000000  female  2007  
2           195.000000  3250.000000  female  2007  
3           200.915205  4201.754386    male  2007  
4           193.000000  3450.000000  female  2007  
..                 ...          ...     ...   ...  
339         207.000000  4000.000000    male  2009  
340         202.000000  3400.000000  female  2009  
341         193.000000  3775.000000    male  2009  
342         210.000000  4100.000000    male  2009  
343         198.000000  3775.000000  female  2009  

[344 rows x 9 columns]

In [308]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['year'] = penguins['year'].astype('category')

In [309]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    int64   
 1   species            344 non-null    category
 2   island             344 non-null    category
 3   bill_length_mm     344 non-null    float64 
 4   bill_depth_mm      344 non-null    float64 
 5   flipper_length_mm  344 non-null    float64 
 6   body_mass_g        344 non-null    float64 
 7   sex                344 non-null    category
 8   year               344 non-null    category
dtypes: category(4), float64(4), int64(1)
memory usage: 15.4 KB


In [310]:
penguins.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [311]:
penguins.describe()

rowid  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count  344.000000      344.000000     344.000000         344.000000   
mean   172.500000       43.921930      17.151170         200.915205   
std     99.448479        5.443643       1.969027          14.020657   
min      1.000000       32.100000      13.100000         172.000000   
25%     86.750000       39.275000      15.600000         190.000000   
50%    172.500000       44.250000      17.300000         197.000000   
75%    258.250000       48.500000      18.700000         213.000000   
max    344.000000       59.600000      21.500000         231.000000   

       body_mass_g  
count   344.000000  
mean   4201.754386  
std     799.613058  
min    2700.000000  
25%    3550.000000  
50%    4050.000000  
75%    4750.000000  
max    6300.000000

<h2>EXAMPLE QUESTION AND INSIGHTS</h2>

<h3>Which species has the heaviest body mass?</h3>

In [312]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [313]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']

In [314]:
gentoo_df['body_mass_g'].mean()

np.float64(5068.965761177136)

In [315]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']
chinstrap_df

rowid    species island  bill_length_mm  bill_depth_mm  \
276    277  Chinstrap  Dream            46.5           17.9   
277    278  Chinstrap  Dream            50.0           19.5   
278    279  Chinstrap  Dream            51.3           19.2   
279    280  Chinstrap  Dream            45.4           18.7   
280    281  Chinstrap  Dream            52.7           19.8   
..     ...        ...    ...             ...            ...   
339    340  Chinstrap  Dream            55.8           19.8   
340    341  Chinstrap  Dream            43.5           18.1   
341    342  Chinstrap  Dream            49.6           18.2   
342    343  Chinstrap  Dream            50.8           19.0   
343    344  Chinstrap  Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
276              192.0       3500.0  female  2007  
277              196.0       3900.0    male  2007  
278              193.0       3650.0    male  2007  
279              188.0       3525.0  female  2007  
280              197.0       3725.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[68 rows x 9 columns]

In [316]:
chinstrap_df['body_mass_g'].mean()

np.float64(3733.0882352941176)

In [317]:
adelie_df = penguins[penguins['species'] == 'Adelie']

In [318]:
adelie_df['body_mass_g'].mean()

np.float64(3703.95891043398)

<h4> INSIGHTS: Gentoo has the heaviest average body mass</h4>

<h2> HOW TO CREATE YOUR COLUMNS </h2>

In [319]:
penguins.columns

Index(['rowid', 'species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [320]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    int64   
 1   species            344 non-null    category
 2   island             344 non-null    category
 3   bill_length_mm     344 non-null    float64 
 4   bill_depth_mm      344 non-null    float64 
 5   flipper_length_mm  344 non-null    float64 
 6   body_mass_g        344 non-null    float64 
 7   sex                344 non-null    category
 8   year               344 non-null    category
dtypes: category(4), float64(4), int64(1)
memory usage: 15.4 KB


In [321]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] /1000,1)

In [322]:
penguins.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
mass_kg              0
dtype: int64

<h1>SELF-EXPLANATORY DATA ANALYTICS</h1>

<h2>QUESTION #1</h2>

<h2>Which among the male and female species has the highest number of records on year 2007?</h2>
<hr>

<h2>Male</h2>

In [323]:
penguins[(penguins['year'] == 2007) & (penguins['sex'] == 'male')]

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
3        4     Adelie  Torgersen        43.92193       17.15117   
5        6     Adelie  Torgersen        39.30000       20.60000   
7        8     Adelie  Torgersen        39.20000       19.60000   
8        9     Adelie  Torgersen        34.10000       18.10000   
9       10     Adelie  Torgersen        42.00000       20.20000   
10      11     Adelie  Torgersen        37.80000       17.10000   
11      12     Adelie  Torgersen        37.80000       17.30000   
13      14     Adelie  Torgersen        38.60000       21.20000   
14      15     Adelie  Torgersen        34.60000       21.10000   
17      18     Adelie  Torgersen        42.50000       20.70000   
19      20     Adelie  Torgersen        46.00000       21.50000   
21      22     Adelie     Biscoe        37.70000       18.70000   
23      24     Adelie     Biscoe        38.20000       18.10000   
24      25     Adelie     Biscoe        38.80000       17.20000   
26      27     Adelie     Biscoe        40.60000       18.60000   
29      30     Adelie     Biscoe        40.50000       18.90000   
31      32     Adelie      Dream        37.20000       18.10000   
33      34     Adelie      Dream        40.90000       18.90000   
35      36     Adelie      Dream        39.20000       21.10000   
36      37     Adelie      Dream        38.80000       20.00000   
39      40     Adelie      Dream        39.80000       19.10000   
41      42     Adelie      Dream        40.80000       18.40000   
43      44     Adelie      Dream        44.10000       19.70000   
45      46     Adelie      Dream        39.60000       18.80000   
46      47     Adelie      Dream        41.10000       19.00000   
47      48     Adelie      Dream        37.50000       18.90000   
49      50     Adelie      Dream        42.30000       21.20000   
153    154     Gentoo     Biscoe        50.00000       16.30000   
155    156     Gentoo     Biscoe        50.00000       15.20000   
156    157     Gentoo     Biscoe        47.60000       14.50000   
159    160     Gentoo     Biscoe        46.70000       15.30000   
161    162     Gentoo     Biscoe        46.80000       15.40000   
163    164     Gentoo     Biscoe        49.00000       16.10000   
165    166     Gentoo     Biscoe        48.40000       14.60000   
167    168     Gentoo     Biscoe        49.30000       15.70000   
169    170     Gentoo     Biscoe        49.20000       15.20000   
171    172     Gentoo     Biscoe        48.70000       15.10000   
172    173     Gentoo     Biscoe        50.20000       14.30000   
175    176     Gentoo     Biscoe        46.30000       15.80000   
177    178     Gentoo     Biscoe        46.10000       15.10000   
178    179     Gentoo     Biscoe        44.50000       14.30000   
179    180     Gentoo     Biscoe        47.80000       15.00000   
181    182     Gentoo     Biscoe        50.00000       15.30000   
182    183     Gentoo     Biscoe        47.30000       15.30000   
185    186     Gentoo     Biscoe        59.60000       17.00000   
277    278  Chinstrap      Dream        50.00000       19.50000   
278    279  Chinstrap      Dream        51.30000       19.20000   
280    281  Chinstrap      Dream        52.70000       19.80000   
283    284  Chinstrap      Dream        51.30000       18.20000   
285    286  Chinstrap      Dream        51.30000       19.90000   
287    288  Chinstrap      Dream        51.70000       20.30000   
289    290  Chinstrap      Dream        52.00000       18.10000   
291    292  Chinstrap      Dream        50.50000       19.60000   
292    293  Chinstrap      Dream        50.30000       20.00000   
295    296  Chinstrap      Dream        49.20000       18.20000   
297    298  Chinstrap      Dream        48.50000       17.50000   
299    300  Chinstrap      Dream        50.60000       19.40000   
301    302  Chinstrap      Dream        52.00000    

<h2>Female</h2>

In [324]:
penguins[(penguins['year'] == 2007) & (penguins['sex'] == 'female')]

rowid    species     island  bill_length_mm  bill_depth_mm  \
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
4        5     Adelie  Torgersen            36.7           19.3   
6        7     Adelie  Torgersen            38.9           17.8   
12      13     Adelie  Torgersen            41.1           17.6   
15      16     Adelie  Torgersen            36.6           17.8   
16      17     Adelie  Torgersen            38.7           19.0   
18      19     Adelie  Torgersen            34.4           18.4   
20      21     Adelie     Biscoe            37.8           18.3   
22      23     Adelie     Biscoe            35.9           19.2   
25      26     Adelie     Biscoe            35.3           18.9   
27      28     Adelie     Biscoe            40.5           17.9   
28      29     Adelie     Biscoe            37.9           18.6   
30      31     Adelie      Dream            39.5           16.7   
32      33     Adelie      Dream            39.5           17.8   
34      35     Adelie      Dream            36.4           17.0   
37      38     Adelie      Dream            42.2           18.5   
38      39     Adelie      Dream            37.6           19.3   
40      41     Adelie      Dream            36.5           18.0   
42      43     Adelie      Dream            36.0           18.5   
44      45     Adelie      Dream            37.0           16.9   
48      49     Adelie      Dream            36.0           17.9   
152    153     Gentoo     Biscoe            46.1           13.2   
154    155     Gentoo     Biscoe            48.7           14.1   
157    158     Gentoo     Biscoe            46.5           13.5   
158    159     Gentoo     Biscoe            45.4           14.6   
160    161     Gentoo     Biscoe            43.3           13.4   
162    163     Gentoo     Biscoe            40.9           13.7   
164    165     Gentoo     Biscoe            45.5           13.7   
166    167     Gentoo     Biscoe            45.8           14.6   
168    169     Gentoo     Biscoe            42.0           13.5   
170    171     Gentoo     Biscoe            46.2           14.5   
173    174     Gentoo     Biscoe            45.1           14.5   
174    175     Gentoo     Biscoe            46.5           14.5   
176    177     Gentoo     Biscoe            42.9           13.1   
180    181     Gentoo     Biscoe            48.2           14.3   
183    184     Gentoo     Biscoe            42.8           14.2   
184    185     Gentoo     Biscoe            45.1           14.5   
276    277  Chinstrap      Dream            46.5           17.9   
279    280  Chinstrap      Dream            45.4           18.7   
281    282  Chinstrap      Dream            45.2           17.8   
282    283  Chinstrap      Dream            46.1           18.2   
284    285  Chinstrap      Dream            46.0           18.9   
286    287  Chinstrap      Dream            46.6           17.8   
288    289  Chinstrap      Dream            47.0           17.3   
290    291  Chinstrap      Dream            45.9           17.1   
293    294  Chinstrap      Dream            58.0           17.8   
294    295  Chinstrap      Dream            46.4           18.6   
296    297  Chinstrap      Dream            42.4           17.3   
298    299  Chinstrap      Dream            43.2           16.6   
300    301  Chinstrap      Dream            46.7           17.9   

     flipper_length_mm  body_mass_g     sex  year  mass_kg  
1                186.0       3800.0  female  2007      3.8  
2                195.0       3250.0  female  2007      3.2  
4                193.0       3450.0  female  2007      3.4  
6                181.0       3625.0  female  2007      3.6  
12               182.0       3200.0  female  2007      3.2  
15               185.0       3700.0  female  2007      3.7  
16               195.0       3450.0  female  2007      3.4  
18               184.0       332

<h2>Results of Male</h2>

In [325]:
male_2007_count = ((penguins['year'] == 2007) & (penguins['sex'] == 'male')).sum()
male_2007_count

np.int64(59)

<h2>Results of Female</h2>

In [326]:
female_2007_count = ((penguins['year'] == 2007) & (penguins['sex'] == 'female')).sum()
female_2007_count

np.int64(51)

<h3>INSIGHTS: </h3>

We show in the given results that the highest record in the year 2007 was the male count, which was 59, and then the female count was 51.


<hr>
<h2>QUESTION #2</h2>

<h2>How many penguins in the palmer penguins database weigh less than 3500 grams?</h2>

In [327]:
penguins[penguins['body_mass_g'] <= 3500].value_counts('body_mass_g')

body_mass_g
3400.0    8
3450.0    8
3500.0    7
3300.0    6
3200.0    5
3250.0    5
3325.0    5
3350.0    5
3150.0    4
3050.0    4
2900.0    4
3475.0    3
2850.0    2
3000.0    2
3425.0    2
3175.0    2
2700.0    1
2925.0    1
2975.0    1
3275.0    1
3075.0    1
3100.0    1
Name: count, dtype: int64

In [328]:
peguins_heavy = penguins[penguins['body_mass_g'] <= 3500].value_counts('body_mass_g')

In [329]:
peguins_heavy.sum()

np.int64(78)

<h3>INSIGHTS: </h3>

The analysis shows that there are 78 penguins in the Palmer Penguins database with a body mass of 3500 grams or less. The code provides a detailed breakdown of the number of penguins at specific weights and confirms the total count.

<hr>
<h2>QUESTION #3</h2>

<h2>Which penguins in the database have a bill length that is more than 50mm, a bill depth that is more than 40mm?</h2>

In [330]:
penguins[(penguins['bill_length_mm'] > 50) | (penguins['bill_depth_mm'] > 40)]

rowid    species  island  bill_length_mm  bill_depth_mm  \
172    173     Gentoo  Biscoe            50.2           14.3   
185    186     Gentoo  Biscoe            59.6           17.0   
196    197     Gentoo  Biscoe            50.5           15.9   
199    200     Gentoo  Biscoe            50.5           15.9   
205    206     Gentoo  Biscoe            50.1           15.0   
211    212     Gentoo  Biscoe            50.4           15.3   
215    216     Gentoo  Biscoe            54.3           15.7   
221    222     Gentoo  Biscoe            50.7           15.0   
229    230     Gentoo  Biscoe            51.1           16.3   
233    234     Gentoo  Biscoe            52.5           15.6   
237    238     Gentoo  Biscoe            50.8           17.3   
239    240     Gentoo  Biscoe            51.3           14.2   
241    242     Gentoo  Biscoe            52.1           17.0   
243    244     Gentoo  Biscoe            52.2           17.1   
247    248     Gentoo  Biscoe            50.8           15.7   
251    252     Gentoo  Biscoe            51.1           16.5   
253    254     Gentoo  Biscoe            55.9           17.0   
259    260     Gentoo  Biscoe            53.4           15.8   
262    263     Gentoo  Biscoe            50.5           15.2   
265    266     Gentoo  Biscoe            51.5           16.3   
267    268     Gentoo  Biscoe            55.1           16.0   
273    274     Gentoo  Biscoe            50.4           15.7   
278    279  Chinstrap   Dream            51.3           19.2   
280    281  Chinstrap   Dream            52.7           19.8   
283    284  Chinstrap   Dream            51.3           18.2   
285    286  Chinstrap   Dream            51.3           19.9   
287    288  Chinstrap   Dream            51.7           20.3   
289    290  Chinstrap   Dream            52.0           18.1   
291    292  Chinstrap   Dream            50.5           19.6   
292    293  Chinstrap   Dream            50.3           20.0   
293    294  Chinstrap   Dream            58.0           17.8   
299    300  Chinstrap   Dream            50.6           19.4   
301    302  Chinstrap   Dream            52.0           19.0   
302    303  Chinstrap   Dream            50.5           18.4   
305    306  Chinstrap   Dream            52.8           20.0   
307    308  Chinstrap   Dream            54.2           20.8   
309    310  Chinstrap   Dream            51.0           18.8   
313    314  Chinstrap   Dream            52.0           20.7   
315    316  Chinstrap   Dream            53.5           19.9   
318    319  Chinstrap   Dream            50.9           19.1   
320    321  Chinstrap   Dream            50.9           17.9   
321    322  Chinstrap   Dream            50.8           18.5   
322    323  Chinstrap   Dream            50.1           17.9   
324    325  Chinstrap   Dream            51.5           18.7   
327    328  Chinstrap   Dream            51.4           19.0   
329    330  Chinstrap   Dream            50.7           19.7   
331    332  Chinstrap   Dream            52.2           18.8   
334    335  Chinstrap   Dream            50.2           18.8   
336    337  Chinstrap   Dream            51.9           19.5   
339    340  Chinstrap   Dream            55.8           19.8   
342    343  Chinstrap   Dream            50.8           19.0   
343    344  Chinstrap   Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  mass_kg  
172              218.0       5700.0    male  2007      5.7  
185              230.0       6050.0    male  2007      6.0  
196              222.0       5550.0    male  2008      5.6  
199              225.0       5400.0    male  2008      5.4  
205              225.0       5000.0    male  2008      5.0  
211              224.0       5550.0    male  2008      5.6  
215              231.0       5650.0    male  2008      5.6  
221              223.0       5550.0    male  2008      5.6  
229              220.0       6000.0    male  2008      6.0  
23

In [331]:
big_bill_penguins = penguins[(penguins['bill_length_mm'] > 50) | (penguins['bill_depth_mm'] > 40)]
big_bill_penguins['species'].value_counts()

species
Chinstrap    30
Gentoo       22
Adelie        0
Name: count, dtype: int64

<h3>INSIGHTS: </h3>

Based on the result provided, the penguins that meet these criteria are from the Chinstrap and Gentoo species. There are 30 Chinstrap penguins with a bill length greater than 50mm or a bill depth greater than 40mm. There are 22 Gentoo penguins that meet the same criteria. There are no Adelie penguins that meet these conditions.

<hr>
<h2>QUESTION #4</h2>

<h2>What is the average length of the flipper of a penguin and define each species?</h2>

In [332]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [333]:
avg_species =penguins.groupby(['species'])['flipper_length_mm'].mean()

In [334]:
avg_species

species
Adelie       190.025758
Chinstrap    195.823529
Gentoo       217.055768
Name: flipper_length_mm, dtype: float64

<h3> INSIGHTS: </h3>

Based on the result provided, While the overall average is around 200.9 mm, there are notable differences when looking at individual species. Gentoo penguins have the longest flippers, with an average length of 217 mm. In contrast, Adelie penguins have the shortest flippers, averaging 190 mm, while Chinstrap penguins have an average of 196 mm.

<hr>
<h2>QUESTION #5</h2>

<h2>What is the typical mass of a Gentoo penguin in kilogram?</h2>

In [335]:
penguins[penguins['species'] == 'Gentoo']['mass_kg'].max()

np.float64(6.3)

In [336]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']
gentoo_df

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152    153  Gentoo  Biscoe        46.10000       13.20000         211.000000   
153    154  Gentoo  Biscoe        50.00000       16.30000         230.000000   
154    155  Gentoo  Biscoe        48.70000       14.10000         210.000000   
155    156  Gentoo  Biscoe        50.00000       15.20000         218.000000   
156    157  Gentoo  Biscoe        47.60000       14.50000         215.000000   
..     ...     ...     ...             ...            ...                ...   
271    272  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272    273  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273    274  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274    275  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275    276  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year  mass_kg  
152  4500.000000  female  2007      4.5  
153  5700.000000    male  2007      5.7  
154  4450.000000  female  2007      4.4  
155  5700.000000    male  2007      5.7  
156  5400.000000    male  2007      5.4  
..           ...     ...   ...      ...  
271  4201.754386    male  2009      4.2  
272  4850.000000  female  2009      4.8  
273  5750.000000    male  2009      5.8  
274  5200.000000  female  2009      5.2  
275  5400.000000    male  2009      5.4  

[124 rows x 10 columns]

<h3> INSIGHTS: </h3>

Based on the result provided, the typical mass of an Gentoo penguin was 6.3kg.

<hr>
<h2>QUESTION #6</h2>

<h2>Which of the Three Islands Hosts the Largest Penguin Population?</h2>

In [337]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h3>INSIGHTS: </h3>

Based on the result, Biscoe Island is the most largest penguin population, with 168 penguins. This is much higher than Dream Island’s 124 penguins and Torgersen Island’s 52 penguins. Each island hosts different species, but Biscoe stands out with the biggest overall number of penguins.

<hr>
<h2>QUESTION #7</h2>

<h2>What is the average mass in kilograms per species?</h2>

<h3>Average of Adelie</h3>

In [338]:
penguins[penguins['species'] == 'Adelie']['mass_kg'].mean().round()

np.float64(4.0)

<h3>Average of Chinstrap</h3>

In [339]:
penguins[penguins['species'] == 'Chinstrap']['mass_kg'].mean().round()

np.float64(4.0)

<h3>Average of Gentoo</h3>

In [340]:
penguins[penguins['species'] == 'Gentoo']['mass_kg'].mean().round()

np.float64(5.0)

<h3>INSIGHTS:</h3>

Gentoo penguins are the heaviest, weighing about 5 kg on average. Adelie and Chinstrap penguins both weigh around 4 kg. Some Adelie and Chinstrap penguins are smaller, weighing 3.5 kg or less.


<hr>
<h2>QUESTION #8</h2>

<h2>On year 2007-2009 which among the three species are in the middle of the shortest and the lowest?</h2>
<hr>

In [341]:
species_year = penguins.groupby(['species','year']).size()
species_year

species    year
Adelie     2007    50
           2008    50
           2009    52
Chinstrap  2007    26
           2008    18
           2009    24
Gentoo     2007    34
           2008    46
           2009    44
dtype: int64

In [342]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h3>INSIGHTS:</h3>

Based on the results, The Gentoo penguin has a total count of 124, placing it in the middle between the Chinstrap and Adelie species, with counts consistently in the middle range each year. The Chinstrap has the lowest total count at 68 and consistently records the lowest count each year. In contrast, the Adelie penguin has the highest total count at 152 and consistently has the highest count each year.

<hr>
<h2>QUESTION #9</h2>

<h3>Is there a difference in mass between male and female penguins at each island?</h3>

In [343]:
mass_kg = penguins.groupby(['species','sex','island'])['mass_kg'].mean()
mass_kg 

species    sex     island   
Adelie     female  Biscoe       3.372727
                   Dream        3.344444
                   Torgersen    3.387500
           male    Biscoe       4.059091
                   Dream        4.013793
                   Torgersen    3.989286
Chinstrap  female  Biscoe            NaN
                   Dream        3.523529
                   Torgersen         NaN
           male    Biscoe            NaN
                   Dream        3.938235
                   Torgersen         NaN
Gentoo     female  Biscoe       4.681034
                   Dream             NaN
                   Torgersen         NaN
           male    Biscoe       5.410606
                   Dream             NaN
                   Torgersen         NaN
Name: mass_kg, dtype: float64

<h3>INSIGHTS:</h3>

Based on the results, Males of the Adelie penguin species are heavier than females across all islands, while there is insufficient data for Chinstrap penguins to compare male and female mass, and for Gentoo penguins, males appear heavier on Biscoe Island, with no data available for other islands.

<hr>
<h2>QUESTION #10</h2>

<h3>Which of the male and female penguins has the shortest length of a bill?</h3>

In [344]:
sex_islands = penguins.groupby(['sex','island','bill_length_mm']).size()
sex_islands

sex     island     bill_length_mm
female  Biscoe     32.1              0
                   33.1              0
                   33.5              0
                   34.0              0
                   34.1              0
                                    ..
male    Torgersen  55.1              0
                   55.8              0
                   55.9              0
                   58.0              0
                   59.6              0
Length: 990, dtype: int64

<h3>INSIGHTS:</h3>

Based on the result provided, The female penguin has the shortest bill length recorded at 32.1 mm then the Male penguins have significantly longer bill lengths than the female penguins, with their shortest recorded bill length starting at 55.1 mm.


<hr>
<h2>QUESTION #11</h2>

<h3>Which species has the widest bill of an penguin?</h3>

In [345]:
avg_species = penguins.groupby(['species'])['bill_depth_mm'].mean()
avg_species 

species
Adelie       18.338495
Chinstrap    18.420588
Gentoo       14.999606
Name: bill_depth_mm, dtype: float64

<h3>INSIGHTS:</h3>

The Chinstrap penguin has the widest bill depth at 18.4 mm, then Adelie follows closely with an average of 18.3 mm.then the Gentoo penguin has a shortest bill depth at 14.9 mm.


<hr>
<h2>QUESTION #12</h2>

<h2>In which particular year are there the fewest records found on the island</h2>

In [346]:
year_island = penguins.groupby(['year','island']).count()
year_island

rowid  species  bill_length_mm  bill_depth_mm  \
year island                                                     
2007 Biscoe        44       44              44             44   
     Dream         46       46              46             46   
     Torgersen     20       20              20             20   
2008 Biscoe        64       64              64             64   
     Dream         34       34              34             34   
     Torgersen     16       16              16             16   
2009 Biscoe        60       60              60             60   
     Dream         44       44              44             44   
     Torgersen     16       16              16             16   

                flipper_length_mm  body_mass_g  sex  mass_kg  
year island                                                   
2007 Biscoe                    44           44   44       44  
     Dream                     46           46   46       46  
     Torgersen                 20           20   20       20  
2008 Biscoe                    64           64   64       64  
     Dream                     34           34   34       34  
     Torgersen                 16           16   16       16  
2009 Biscoe                    60           60   60       60  
     Dream                     44           44   44       44  
     Torgersen                 16           16   16       16

In [347]:
penguins['year'].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h3>INSIGHTS:</h3>

In 2007, Torgersen Island recorded the fewest total records at 20. However, both 2008 and 2009 also saw lower counts on Torgersen, each with only 16 records.

<hr>
<h2>QUESTION #13</h2>

<h3>What does the p-value indicate regarding the difference in flipper length between male and female penguins?</h3>

In [348]:
stats, p_value = f_oneway(penguins[penguins['sex'] == 'male']['flipper_length_mm'],penguins[penguins['sex'] == 'female']['flipper_length_mm'])

In [349]:
p_value

np.float64(4.88065474065626e-06)

<h3>INSIGHTS:</h3>

Given the extremely low p-value, we can confidently say that male and female penguins have significantly different flipper lengths.

<hr>

<h2>DATA AGGREGATION</h2>

<h3>1.Ilan species ng penguin ang nakatira per island?</h3>

In [350]:
species_island = penguins.groupby(['species','island']).count()

In [351]:
species_island

rowid  bill_length_mm  bill_depth_mm  flipper_length_mm  \
species   island                                                               
Adelie    Biscoe        44              44             44                 44   
          Dream         56              56             56                 56   
          Torgersen     52              52             52                 52   
Chinstrap Biscoe         0               0              0                  0   
          Dream         68              68             68                 68   
          Torgersen      0               0              0                  0   
Gentoo    Biscoe       124             124            124                124   
          Dream          0               0              0                  0   
          Torgersen      0               0              0                  0   

                     body_mass_g  sex  year  mass_kg  
species   island                                      
Adelie    Biscoe              44   44    44       44  
          Dream               56   56    56       56  
          Torgersen           52   52    52       52  
Chinstrap Biscoe               0    0     0        0  
          Dream               68   68    68       68  
          Torgersen            0    0     0        0  
Gentoo    Biscoe             124  124   124      124  
          Dream                0    0     0        0  
          Torgersen            0    0     0        0

In [352]:
species_islands = penguins.groupby(['species','island']).size()

In [353]:
species_islands

species    island   
Adelie     Biscoe        44
           Dream         56
           Torgersen     52
Chinstrap  Biscoe         0
           Dream         68
           Torgersen      0
Gentoo     Biscoe       124
           Dream          0
           Torgersen      0
dtype: int64

In [354]:
species_islands1 = penguins.groupby(['species','island','sex']).size()

In [355]:
species_islands1

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

<h3>2. average weight of penguin per island and sex?</h3>

In [356]:
avg_island_s =penguins.groupby(['species','island','sex'])['body_mass_g'].mean()

In [357]:
avg_island_s

species    island     sex   
Adelie     Biscoe     female    3369.318182
                      male      4050.000000
           Dream      female    3344.444444
                      male      4008.620690
           Torgersen  female    3395.833333
                      male      3990.241228
Chinstrap  Biscoe     female            NaN
                      male              NaN
           Dream      female    3527.205882
                      male      3938.970588
           Torgersen  female            NaN
                      male              NaN
Gentoo     Biscoe     female    4679.741379
                      male      5411.011430
           Dream      female            NaN
                      male              NaN
           Torgersen  female            NaN
                      male              NaN
Name: body_mass_g, dtype: float64

<h1>CORRELATION ANALYSIS</h1>
<hr>
<h3>Testing Correlation using Pearson R</h3>
<hr>
<p><b>Pearson R</b> can test correlation between two Variables, given that both Variables are <strong>NUMERICAL DATA</strong></p>

In [358]:
penguins.dtypes

rowid                   int64
species              category
island               category
bill_length_mm        float64
bill_depth_mm         float64
flipper_length_mm     float64
body_mass_g           float64
sex                  category
year                 category
mass_kg               float64
dtype: object

In [359]:
flipper_mass = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])

In [360]:
flipper_mass

np.float64(0.8712017673060114)

<h2>Installing Scipy</h2>

In [361]:
!pip install scipy

In [362]:
!python -m pip install --upgrade pip

In [363]:
penguins[penguins['sex'] == 'male']

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
3        4     Adelie  Torgersen        43.92193       17.15117   
5        6     Adelie  Torgersen        39.30000       20.60000   
7        8     Adelie  Torgersen        39.20000       19.60000   
8        9     Adelie  Torgersen        34.10000       18.10000   
..     ...        ...        ...             ...            ...   
334    335  Chinstrap      Dream        50.20000       18.80000   
336    337  Chinstrap      Dream        51.90000       19.50000   
339    340  Chinstrap      Dream        55.80000       19.80000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   

     flipper_length_mm  body_mass_g   sex  year  mass_kg  
0           181.000000  3750.000000  male  2007      3.8  
3           200.915205  4201.754386  male  2007      4.2  
5           190.000000  3650.000000  male  2007      3.6  
7           195.000000  4675.000000  male  2007      4.7  
8           193.000000  3475.000000  male  2007      3.5  
..                 ...          ...   ...   ...      ...  
334         202.000000  3800.000000  male  2009      3.8  
336         206.000000  3950.000000  male  2009      4.0  
339         207.000000  4000.000000  male  2009      4.0  
341         193.000000  3775.000000  male  2009      3.8  
342         210.000000  4100.000000  male  2009      4.1  

[179 rows x 10 columns]

In [364]:
from scipy.stats import f_oneway

In [365]:
penguins[penguins['sex'] == 'male']['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [366]:
stats, p_value = f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])

In [367]:
p_value

np.float64(3.0226600487050246e-15)

In [368]:
flipper_mass = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])

In [369]:
flipper_mass

np.float64(0.8712017673060114)

<hr>

<h1>CATEGORICAL DATA</h1>

<h2>BODY MASS VS SEX</h2>

In [370]:
#import scipy.stats
from scipy.stats import f_oneway

In [371]:
#FILTER DATA
penguins['sex'] == 'male'

0       True
1      False
2      False
3       True
4      False
       ...  
339     True
340    False
341     True
342     True
343    False
Name: sex, Length: 344, dtype: bool

In [372]:
penguins[penguins['sex'] == 'male']['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [373]:
penguins[penguins['sex'] == 'female']['body_mass_g']

1      3800.0
2      3250.0
4      3450.0
6      3625.0
12     3200.0
        ...  
335    3525.0
337    3650.0
338    3650.0
340    3400.0
343    3775.0
Name: body_mass_g, Length: 165, dtype: float64

In [374]:
no_use, p_value = f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'], penguins[penguins['sex'] == 'female']['body_mass_g'])

In [375]:
p_value

np.float64(3.0226600487050246e-15)

<hr>